kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1168 @@
+        // rhs_h =  T'*(rhs - K*g)
+        VectorType temp_vec = ( rhs_contribution - prod(lhs_contribution, constant_vector_local) );
+        noalias(rhs_contribution) = prod( trans(transformation_matrix_local), temp_vec );
+        // lhs_h = T'*K*T
+        MatrixType temp_mat = prod(lhs_contribution, transformation_matrix_local);
+        noalias(lhs_contribution) = prod( trans(transformation_matrix_local),  temp_mat);
+        // rhs_h(s,s) = rhs(s,s) : that is reassigning the slave part of the matrix back. We do not modify the (slave, slave) block
+        // this is to facilitate the solution of the linear system of equation.
+        for (auto &slave_index : local_slave_index_vector)
+            for (auto &slave_index_other : local_slave_index_vector)
+                lhs_contribution(slave_index, slave_index_other) = rLHSContribution(slave_index, slave_index_other);
+        // rhs_h(s,i) = 0 and rhs_h(i,s) = 0
+        // making this blocks zero will ensure that the slaves are not connected to internal dofs
+        for (auto &slave_index : local_slave_index_vector)
+            for (auto &internal_index : local_internal_index_vector)
+            {
+                lhs_contribution(slave_index, internal_index) = 0.0;
+                lhs_contribution(internal_index, slave_index) = 0.0;
+            }
+
+        for (auto &slave_index : local_slave_index_vector)
+            rhs_contribution(slave_index) = 0.0;
+
+        rLHSContribution.resize(rEquationIds.size(), rEquationIds.size());
+        rRHSContribution.resize(rEquationIds.size());
+        noalias(rLHSContribution) = lhs_contribution;
+        noalias(rRHSContribution) = rhs_contribution;
+
+        KRATOS_CATCH("ResidualBasedBlockBuilderAndSolverWithConstraints:: Applying Multipoint constraints failed ..");
+    }
+
+
+
+    /**
+     * @brief   This function resizes the given matrix and vector pair to the new size provided.
+     *          And Initializes the extra part added to zero.
+     * @param   rMatrix matrix to be resized
+     * @param   rVector vector to be resized
+     * @param   FinalSize the final size of the resized quantities.
+     */
+    void ResizeAndInitializeLocalMatrices(MatrixType& rMatrix, VectorType& rVector,
+                                            IndexType FinalSize)
+    {
+        KRATOS_TRY
+        // storing the initial matrix and vector and their properties
+        KRATOS_ERROR_IF(rMatrix.size1() != rVector.size())<<"ResizeAndInitializeLocalMatrices :: Dimension of the matrix and vector passed are not the same !"<<std::endl;
+        const IndexType initial_sys_size = rMatrix.size1();
+        MatrixType matrix(initial_sys_size, initial_sys_size);
+        noalias(matrix) = rMatrix;
+        VectorType vector(initial_sys_size);
+        noalias(vector) = rVector;
+
+        rMatrix.resize(FinalSize, FinalSize, true); //true for Preserving the data and resizing the matrix
```

Comment:
please do not use this feature...

Reply:
@RiccardoRossi  I am formulating without using this feature. !